scripts/hotWalletCli/wallet.ts

```diff
@@ -0,0 +1,336 @@
+
+const suffix = (text: string): string => {
+  return `\n${symbols.error} ${chalk.bold.red(text)}`
+}
+
+export class Wallet {
+  private hdwallet: NativeHDWallet
+
+  constructor(mnemonic: string) {
+    this.hdwallet = new NativeHDWallet({ mnemonic, deviceId: 'hot' })
+  }
+
+  static async new(mnemonic: string): Promise<Wallet> {
+    try {
+      const wallet = new Wallet(mnemonic)
+      const initialized = await wallet.initialize()
+
+      if (!initialized) {
+        error('Failed to initialize hot wallet, exiting.')
+        process.exit(1)
+      }
+
+      const { address, path } = await wallet.getAddress()
+
+      info(`Hot wallet address: ${address} (${path})`)
+
+      return wallet
+    } catch {
+      error('Failed to create hot wallet, exiting.')
+      process.exit(1)
+    }
+  }
+
+  private async initialize(): Promise<boolean | null> {
+    return this.hdwallet.initialize()
+  }
+
+  private async getAddress() {
+    try {
+      const address = await this.hdwallet.thorchainGetAddress({ addressNList })
+
+      if (!address) {
+        error('Failed to get address from hot wallet, exiting.')
+        process.exit(1)
+      }
+
+      return { address, path: BIP32_PATH }
+    } catch {
+      error('Failed to get address from hot wallet, exiting.')
+      process.exit(1)
+    }
+  }
+
+  private async buildFundingTransaction(amount: string, epoch: number) {
+    const { address } = await this.getAddress()
+
+    return {
+      body: {
+        messages: [
+          {
+            '@type': '/types.MsgSend',
+            from_address: SHAPESHIFT_MULTISIG_ADDRESS,
+            to_address: address,
+            amount: [
+              {
+                denom: 'rune',
+                amount,
+              },
+            ],
+          },
+        ],
+        memo: `Fund rFOX rewards distribution - Epoch #${epoch}`,
+        timeout_height: '0',
+        extension_options: [],
+        non_critical_extension_options: [],
+      },
+      auth_info: {
+        signer_infos: [],
+        fee: {
+          amount: [],
+          gas_limit: '0',
+          payer: '',
+          granter: '',
+        },
+      },
+      signatures: [],
+    }
+  }
+
+  async fund(epoch: Epoch) {
+    const { address } = await this.getAddress()
+
+    const distributions = Object.values(epoch.distributionsByStakingAddress)
+
+    const totalDistribution = distributions.reduce((prev, distribution) => {
+      return prev + BigInt(distribution.amount)
+    }, 0n)
+
+    const totalFees = BigInt(2000000) * BigInt(distributions.length)
+
+    const totalAmount = (totalDistribution + totalFees).toString()
+
+    const isFunded = async (interval?: NodeJS.Timeout, spinner?: Ora, resolve?: () => void): Promise<boolean> => {
+      try {
+        const { data } = await axios.get<{ result: { total_count: string } }>(
+          `${THORNODE_URL}/rpc/tx_search?query="transfer.recipient='${address}' AND transfer.amount='${totalAmount}rune'"`,
```

Comment:
not blocking, more just thinking out loud since this would be sep ticket: Do we want to account for the fact that msig signers might prefer to do a small test transaction vs the whole amount.

Reply:
We can account for that. It might be worth asking if that feels like a relevant part of the flow. Considering everything is programattic and generated, the test transaction doesn't _feel_ as relevant, but it's def doable if that is the desired flow. I can even generate a test tx as well, but that feels like a bit more work than worth imo